### PR TITLE
Add govulncheck action

### PR DESCRIPTION
**Pull request by katexochen**
### Proposed change(s)

- Add [govulncheck](https://go.dev/blog/vuln)
- Fix links of e2e badge in README, now points to workflow instead of svg.

### CODE REVIEW

File: .github/workflows/test-govulncheck.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    name: govulncheck
+    runs-on: ubuntu-latest
+    env:
+      GOPRIVATE: github.com/edgelesssys/*
+    steps:
+      - name: Checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - name: Install Dependencies
+        run: sudo apt-get update && sudo apt-get -y install libcryptsetup-dev libvirt-dev
```
Comment (3u13r) [inline .github/workflows/test-govulncheck.yml:29]: Let us check if we need those dependency

**Review by 3u13r — CHANGES_REQUESTED** [review]

File: .github/workflows/test-govulncheck.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  govulncheck:
+    name: govulncheck
+    runs-on: ubuntu-latest
+    env:
+      GOPRIVATE: github.com/edgelesssys/*
```
Comment (thomasten) [inline .github/workflows/test-govulncheck.yml:23]:
```suggestion
```
I guess we don't need this anymore

File: cli/main.go [hunk 3]
```diff
@@ -18,3 +18,5 @@ func main() {
 	}
 	os.Exit(0)
 }
+
+// Trigger.
```
Comment (thomasten) [inline cli/main.go:22]:
```suggestion
```

**Review by thomasten — APPROVED** [review]
❤️

File: .github/workflows/test-govulncheck.yml (hunk 1 above)
Comment (katexochen) [reply .github/workflows/test-govulncheck.yml]: :x:  https://github.com/edgelesssys/constellation/actions/runs/3051730595/jobs/4920302841

**Review by 3u13r — APPROVED** [review]
LGTM

**Pull request merged**